cache_test.go

```diff
@@ -248,7 +248,7 @@ func (_ CacheTests) ResizeOnTheFly() {
 		cache.Set(strconv.Itoa(i), i, time.Minute)
 	}
 	cache.SetMaxSize(3)
-	time.Sleep(time.Millisecond * 10)
+	cache.SyncUpdates()
 	Expect(cache.GetDropped()).To.Equal(2)
```

Comment:
This, and the equivalent line in `LayeredCacheTests.ResizeOnTheFly()`, is where I'm still seeing an intermittent failure (but easily reproducible by just running the tests a bunch of times with `-count`). The failure is `expected 1 to be equal to 2`.

I see why that might be happening... but then I'm not sure why it doesn't _always_ happen. That is— we've specified `ItemsToPrune(1)`. Then when it gets the `setMaxSize` message, and it sees that the cache is now too full, it does `dropped += c.gc()`. And it looks to me like whatever `itemsToPrune` is set to, a single call to `gc()` can only remove up to that number of items. So... how does it ever end up dropping 2, rather than 1? I don't see how sleeping any amount of time would change that, because `gc()` won't get called again until the next time there's a promote, a delete, or another `setMaxSize`.

What am I missing?

Reply:
Ah— it looks like that's addressed by your recent change regarding `itemsToPrune` which I didn't have yet. I'm still not sure how the test was passing prior to that change, but anyway now the intermittent error is gone - I see no failures even with `-count 1000`.